src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -0,0 +1,513 @@
+                                                [](uint32_t a, uint32_t b, int64_t mn, int64_t mx) {
+                                                    return static_cast<int64_t>(uniteHighLow(b, a) % (mx - mn) + mn);
+                                                });
+                    } break;
+                    default: THROW_CPU_NODE_ERR("Unsupported type of RandomUniform: ", m_output_prc.to_string());
+                }
+
+                if (++n == 0) {
+                    counter++;
+                }
+            }
+        };
+
+        parallel_nt(m_threads_num, threadBody);
+    }
+
+    // Calculate counter values for next RandomUniform run.
+    n_state += m_skip_count;
+    if (n_state < m_skip_count) {
+        counter_state++;
+    }
+
+    return { n_state, counter_state };
+}
+
+////////////// STL algo ///////////////
+void RandomUniform::computeStl(void* out, size_t work_amount) {
+    switch (m_output_prc) {
+        case element::f32: {
+            generateData<float, std::uniform_real_distribution<float>>(
+                    std::uniform_real_distribution<float>{m_min_val.f32, m_max_val.f32}, out, work_amount);
+        } break;
+        case element::i32: {
+            generateData<int32_t, std::uniform_int_distribution<int32_t>>(
+                    std::uniform_int_distribution<int32_t>{m_min_val.i32, m_max_val.i32}, out, work_amount);
+        } break;
+        case element::i64: {
+            generateData<int64_t, std::uniform_int_distribution<int64_t>>(
+                    std::uniform_int_distribution<int64_t>{m_min_val.i64, m_max_val.i64}, out, work_amount);
+        } break;
+        case element::f64: {
+            generateData<double, std::uniform_real_distribution<double>>(
+                    std::uniform_real_distribution<double>{m_min_val.f64, m_max_val.f64}, out, work_amount);
+        } break;
+        default:
+            THROW_CPU_NODE_ERR("has unsupported output type: ", m_output_prc);
+    }
+}
+
+template <typename T, typename DISTR_TYPE>
+void RandomUniform::generateData(DISTR_TYPE distribution, void* out, size_t work_amount) {
+    auto dst = reinterpret_cast<T*>(out);
+    for (size_t i = 0; i < work_amount; i++) {
+        *dst = distribution(m_generator);
+        dst++;
+    }
+}
+//////////////////////////////////
+
+void RandomUniform::initEdgeValues(OutputType& dst, const void* src, const element::Type& output_type) {
+#define EL_CASE(E) \
+    case element::E: \
+        dst.E = *reinterpret_cast<const element_type_traits<element::E>::value_type *>(src); \
+        break;
+
+    switch (output_type) {
+        EL_CASE(f32)
+        EL_CASE(f16)
+        EL_CASE(bf16)
+        EL_CASE(i32)
+        EL_CASE(i64)
+        EL_CASE(f64)
+        default:
+            THROW_CPU_NODE_ERR("has unsupported output precision: ", output_type);
+    }
+
+#undef EL_CASE
+}
+
+std::string RandomUniform::getPrimitiveDescriptorType() const {
```

Comment:
Why do we need to override this method?

Reply:
The base fn choses the input precision first, but the output precision is important here.